lib/awful/spawn.lua

```diff
@@ -6,6 +6,61 @@
 -- program after it has been launched.  This requires currently that the
 -- applicaton supports them.
 --
+-- **Understanding clients versus PID versus commands versus class**:
+--
+-- A *process* has *PID*. It can have no, one or many *window*.
+--
+-- A *command* if what is used to start *process*(es). It has no direct relation
+-- with the *process*, *client* or *window*. When a command is executed, it will
+-- usually start a *process* and keep it until it exits. This, however, isn't
+-- always true as some applications use scripts as command and some other use
+-- various single instance mechanism (usually client/server) and merge with
+-- an existing process.
+--
+-- A *client* correspond to a *window*. It is owned by a process. It can have
+-- both a parent and one or many children. A *client* has a *class*, an
+-- *instance* and a *role*. See `client.class`, `client.instance`, `client.role`
+-- and `client.type` for more information about these properties.
+--
+-- **The startup notification protocol**:
+--
+-- The startup notification protocol is an optional specification implemented
+-- by X11 applications to bridge the chain of knowledge between the moment a
+-- program is launched to the moment its window (client) is shown. It can be
+-- found [on the FreeDesktop.org website](https://www.freedesktop.org/wiki/Specifications/startup-notification-spec/).
+--
+-- Awesome has support for the various events part of the protocol, but the most
+-- useful is the identifier, usually called by its `SNID` acronym in the
+-- documentation. It isn't usually necessary to even know it exists, as it is all
+-- done automatically. However, if more control is required, the identifier
+-- can be specified in by an environment variable called `DESKTOP_STARTUP_ID`.
+-- For example, if a command such as:
+--
+--    DESKTOP_STARTUP_ID="something" my_command
+--
+-- Is executed in a terminal, it should (if the program correctly implements the
+-- protocol) results in `c.startup_id` to at least match `something`. In case it
+-- is already used, some implementations will happen some suffixes to the string.
```

Comment:
Huh? Any examples for this?

As far as I remember, the `DESKTOP_STARTUP_ID` should have a special format and contain a timestamp. If there is no timestamp, GTK adds one.
How could anything even check if the startup id is already used?